Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-{
-	name = nullptr;
-}
-
-
-
-Item::~Item()
-{
-	if(name)
-		delete[] name;
-}
-
-
-
-char* Item::getItemName()
-{
-	return name;
-}
-
-
-
-//---------------------Treasure------------------------
-
-
-
-Treasure::Treasure()
-{
-	worth = 0;
-}
-
-
-
-Treasure::~Treasure()
-{
-
-}
-
-
-
-Treasure::Treasure(char const* treasure_Name , int treasure_Worth)
-{
-	name = new char [strlen (treasure_Name) + 1];
-	strcpy_s(name,strlen (treasure_Name) + 1, treasure_Name);
-
-	worth = treasure_Worth;
-}
-
-
-
-int Treasure::getWorth()
-{
-	return worth;
-}
-
-
-
-//---------------------Weapon------------------------
-
-
-
-Weapon::Weapon()
-{
-
-}
-
-
-
-Weapon::~Weapon()
-{
-
-}
-
-
-
-Weapon::Weapon(char const* weapon_Name)
-{
-	name = new char [strlen (weapon_Name) + 1];
-	strcpy_s (name, strlen (weapon_Name) + 1, weapon_Name);
-}
-
-
-
-int Weapon::getWorth()
-{
-	return 0;
-}
-
-
-
-//---------------------Monster------------------------
-
-
-
-Monster::Monster()
-{
-	name = nullptr;
-	is_Alive = true;
-	killing_weapon = nullptr;
-}
-
-
-
-Monster::~Monster()
-{
-	if(name)
-		delete[] name;
-}
-
-
-
-Monster::Monster(char const* monster_Name, Item* weapon)
-{
-	name = new char [strlen(monster_Name) + 1];
-	strcpy_s (name,strlen (monster_Name) + 1, monster_Name);
-
-	is_Alive = true;
-	killing_weapon = weapon;
-}
-
-
-
-Item* Monster::getKillingWeapon()
-{
-	return killing_weapon;
-}
-
-
-
-void Monster::setKillingWeapon(Item* item)
-{
-	killing_weapon = item;
-}
-
-
-
-bool Monster::getLivingState()
-{
-	return is_Alive;
-}
-
-
-
-void Monster::setLivingState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-char* Monster::getMonsterName()
-{
-	return name;
-}
-
-
-
-//---------------------Princess------------------------
-
-
-
-Princess::Princess()
-{
-	is_Alive = true;
-}
-
-
-
-Princess::~Princess()
-{
-
-}
-
-
-
-bool Princess::getLivingState()
-{
-	return is_Alive;
-}
-
-
-
-void Princess::setLivingState(bool currentState)
-{
-	is_Alive = currentState;
-}
-
-
-
-//---------------------Room------------------------
-
-
-
-Room::Room()
-{
-	for (int i = 0; i < 4; ++i)
-	{
-		paths[i] = nullptr;
-	}
-
-	for (int i = 0; i < 5; ++i)
-	{
-		items_Present[i] = nullptr;
-	}
-
-	monster_Present  = nullptr;
-	princess_Present = nullptr;
-
-	description 	 = nullptr;
-	roomNumber 		 = 0;
-}
-
-
-
-Room::~Room()
-{
-	if(description)
-		delete[] description;
-}
-
-
-
-int Room::directionIndex(char const* direction)
-{
-	int i = 0;
-	if(strcmp(direction , "NORTH") == 0)
-		i = 0;
-
-	else if(strcmp (direction, "SOUTH") == 0)
-		i = 1;
-
-	else if(strcmp (direction, "EAST") == 0)
-		i = 2;
-
-	else if(strcmp (direction, "WEST") == 0)
-		i = 3;
-
-	return i;
-}
-
-
-
-char* Room::directionName(int i)
-{
-	char* buffer = new char [10];
-
-	if( i == 0)
-		strcpy_s(buffer,10, "North");
-
-	else if(i == 1)
-		strcpy_s(buffer,10, "South");
-
-	else if(i == 2)
-		strcpy_s(buffer,10, "East");
-
-	else if(i == 3)
-		strcpy_s(buffer,10, "West");
-
-	return buffer;
```

Comment:
huge memory leak. creates new char literally every function call, this gets called many times throughout program.

should be:
```c++
 const char * directionName(){
...
return "North";
}
```
or
```c++
 string directionName(){
...
return "North";
}
```